test/powershell/Language/Parser/TypeAccelerator.Tests.ps1

```diff
@@ -1,25 +1,428 @@
+            }
+            @{
+                Accelerator = 'runspacefactory'
+                Type        = [System.Management.Automation.Runspaces.RunspaceFactory]
+            }
+            @{
+                Accelerator = 'runspace'
+                Type        = [System.Management.Automation.Runspaces.Runspace]
+            }
+            @{
+                Accelerator = 'initialsessionstate'
+                Type        = [System.Management.Automation.Runspaces.InitialSessionState]
+            }
+            @{
+                Accelerator = 'psscriptmethod'
+                Type        = [System.Management.Automation.PSScriptMethod]
+            }  
+            @{
+                Accelerator = 'psscriptproperty'
+                Type        = [System.Management.Automation.PSScriptProperty]
+            }
+            @{
+                Accelerator = 'psnoteproperty'
+                Type        = [System.Management.Automation.PSNoteProperty]
+            }
+            @{
+                Accelerator = 'psaliasproperty'
+                Type        = [System.Management.Automation.PSAliasProperty]
+            }
+            @{
+                Accelerator = 'psvariableproperty'
+                Type        = [System.Management.Automation.PSVariableProperty]
+            }
+        )
+        
+        if ( $IsCoreCLR )
+        {
+            $totalAccelerators = 80 
+        } 
+        else
+        {
+            $totalAccelerators = 82
+
+            $nonCoreTypeAcceleratorTestCases = @(
+                @{
+                    Accelerator = 'adsi'
+                    Type        = [System.DirectoryServices.DirectoryEntry]
+                }
+                @{
+                    Accelerator = 'adsisearcher'
+                    Type        = [System.DirectoryServices.DirectorySearcher]
+                }
+                @{
+                    Accelerator = 'wmiclass'
+                    Type        = [System.Management.ManagementClass]
+                }
+                @{
+                    Accelerator = 'wmi'
+                    Type        = [System.Management.ManagementObject]
+                }
+                @{
+                    Accelerator = 'wmisearcher'
+                    Type        = [System.Management.ManagementObjectSearcher]
+                }
+            )
+        }
+    }
 
     It "Basic type accelerator usage" {
-        [msft_2174855] | Should Be ([int])
+        [userDefinedAcceleratorType] | Should Be ([int])
     }
 
     It "Can query type accelerators" {
-        if ( $IsCoreCLR ) { $count = 80 } else { $count = 82 }
-        $TypeAccelerators.Count -gt $count | Should Be $true
+        $TypeAccelerators.Count -gt $totalAccelerators | Should Be $true
         $TypeAccelerators['xml'] | Should Be ([System.Xml.XmlDocument])
         $TypeAccelerators['AllowNull'] | Should Be ([System.Management.Automation.AllowNullAttribute])
     }
 
     It "Can remove type accelerator" {

```

Comment:
I recommend to group this test and `It "Basic type accelerator usage"` together in a `Context`, and then move the initialization of `userDefinedAcceleratorType` and `userDefinedAcceleratorTypeToRemove` to the `BeforeAll` of the `Context`. And remember to clean them up in `AfterAll` block of the same `Context`.